packages/connect-voting-disputable/subgraph/src/DisputableVoting.ts

```diff
@@ -0,0 +1,213 @@
+import { BigInt, Address } from '@graphprotocol/graph-ts'
+import { ERC20 as ERC20Contract } from '../generated/templates/DisputableVoting/ERC20'
+import {
+  DisputableVoting as DisputableVotingEntity,
+  Setting as SettingEntity,
+  Vote as VoteEntity,
+  CastVote as CastVoteEntity,
+  Voter as VoterEntity,
+  ERC20 as ERC20Entity
+} from '../generated/schema'
+import {
+  DisputableVoting as VotingContract,
+  NewSetting as NewSettingEvent,
+  StartVote as StartVoteEvent,
+  CastVote as CastVoteEvent,
+  PauseVote as PauseVoteEvent,
+  ResumeVote as ResumeVoteEvent,
+  CancelVote as CancelVoteEvent,
+  ExecuteVote as ExecuteVoteEvent,
+  VoteQuietEndingExtension as VoteQuietEndingExtensionEvent,
+  ChangeRepresentative as ChangeRepresentativeEvent
+} from '../generated/templates/DisputableVoting/DisputableVoting'
+
+/* eslint-disable @typescript-eslint/no-use-before-define */
+
+export function handleNewSetting(event: NewSettingEvent): void {
+  const votingApp = VotingContract.bind(event.address)
+  const settingData = votingApp.getSetting(event.params.settingId)
+
+  const voting = loadOrCreateVoting(event.address)
+  const currentSettingId = buildSettingId(event.address, event.params.settingId)
+  const setting = new SettingEntity(currentSettingId)
+  setting.voting = event.address.toHexString()
+  setting.settingId = event.params.settingId
+  setting.supportRequiredPct = settingData.value0
+  setting.minimumAcceptanceQuorumPct = settingData.value1
+  setting.executionDelay = settingData.value2
+  setting.overruleWindow = settingData.value3
+  setting.quietEndingPeriod = settingData.value4
+  setting.quietEndingExtension = settingData.value5
+  setting.createdAt = event.block.timestamp
+  setting.save()
+
+  voting.setting = currentSettingId
+  voting.save()
+}
+
+export function handleStartVote(event: StartVoteEvent): void {
+  const voteId = buildVoteId(event.address, event.params.voteId)
+  const votingApp = VotingContract.bind(event.address)
+
+  const vote = new VoteEntity(voteId)
+  const voteData = votingApp.getVote(event.params.voteId)
+  vote.voting = event.address.toHexString()
+  vote.voteId = event.params.voteId
+  vote.creator = event.params.creator
+  vote.context = event.params.context.toString()
+  vote.yeas = voteData.value0
+  vote.nays = voteData.value1
+  vote.votingPower = voteData.value2
+  vote.setting = buildSettingId(event.address, voteData.value3)
+  vote.actionId = voteData.value4
+  vote.status = castVoteStatus(voteData.value5)
+  vote.startDate = voteData.value6
+  vote.snapshotBlock = voteData.value7
+  vote.pausedAt = voteData.value8
+  vote.pauseDuration = voteData.value9
+  vote.quietEndingExtendedSeconds = voteData.value10
+  vote.quietEndingSnapshotSupport = castVoterState(voteData.value11)
+  vote.script = voteData.value12
```

Comment:
I wonder if the latest version of the contract is here: https://github.com/aragon/aragon-apps/blob/disputable_voting_deploy/apps/voting-disputable/contracts/DisputableVoting.sol#L85

I notice the `voteData` maybe have small differences.

Reply:
Yup, I double-checked and it is that one :)
We are also checking it in the tests